risc0/build/src/lib.rs

```diff
@@ -362,21 +202,13 @@ fn build_guest_package<P>(
     }
 
     fs::create_dir_all(target_dir.as_ref()).unwrap();
-    let cargo = env::var("CARGO").unwrap();
-    let mut std_parts = vec!["alloc", "core", "proc_macro", "panic_abort"];
-    if std {
-        std_parts.push("std");
-    }
-    let build_std = format!("build-std={}", std_parts.join(","));
+
     let mut args = vec![
+        "+risc0",
         "build",
         "--release",
         "--target",
-        guest_build_env.target_spec.to_str().unwrap(),
-        "-Z",
-        build_std.as_str(),
-        "-Z",
-        "build-std-features=compiler-builtins-mem",
+        "riscv32im-risc0-zkvm-elf",
```

Comment:
By dropping these -Z's what are the remaining bits keeping us on nightly?

Reply:
We need to get a build of `cargo` which is independent of the host toolchain.